src/webgpu/gpu_test.ts

```diff
@@ -1467,14 +1500,27 @@ export function TextureTestMixin<F extends FixtureClass<GPUTest>>(
                   resource: texture.createView({
                     baseMipLevel: mipLevel,
                     mipLevelCount: 1,
-                    baseArrayLayer: layer,
-                    arrayLayerCount: 1,
-                    dimension: '2d',
+                    ...(!this.isCompatibility && {
+                      baseArrayLayer: layer,
+                      arrayLayerCount: 1,
+                    }),
+                    dimension: pipelineType as GPUTextureViewDimension,
                   }),
                 },
+                ...(pipelineType === '2d-array'
+                  ? [
+                      {
+                        binding: 2,
+                        resource: { buffer: uniformBuffer },
+                      },
+                    ]
+                  : []),
               ],
             });
 
+            uniformData[0] = layer;
+            this.device.queue.writeBuffer(uniformBuffer, 0, uniformData);
```

Comment:
nit: just initialize uniformData as `new Uint32Array([layer])`, or at least move the definition down here so it's closer and the type is easy to see).

Reply:
I thought we were trying to avoid micro allocations? If I move it down here it happens per iteration instead of per test. I suppose for this test it doesn't matter?